src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -362,15 +362,26 @@ private int FindEntry(TKey key)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.key);
             }
 
-            if (_buckets != null)
+            int[] buckets = _buckets;
+            int i = -1;
+            if (buckets != null)
             {
-                int hashCode = _comparer.GetHashCode(key) & 0x7FFFFFFF;
-                for (int i = _buckets[hashCode % _buckets.Length]; i >= 0; i = _entries[i].next)
+                IEqualityComparer<TKey> comparer = _comparer;
+                int hashCode = comparer.GetHashCode(key) & 0x7FFFFFFF;
+                i = buckets[hashCode % buckets.Length];
+
+                Entry[] entries = _entries;
+                while (i >= 0)
                 {
-                    if (_entries[i].hashCode == hashCode && _comparer.Equals(_entries[i].key, key)) return i;
+                    if (entries[i].hashCode == hashCode && comparer.Equals(key, entries[i].key))
```

Comment:
It is likely that there is (fragile) code out there that depends on the exact order of arguments passed into the comparer. Switching the order is going to break it. Could you please keep preserve the argument order?

Reply:
Switched though it increases code size. However, have found a pattern to drop the range check which makes up for it